### PR TITLE
Rustfmt libterm

### PR DESCRIPTION
**Pull request by SingingTree**
Hey hey,

This is the result of running rustfmt over the libterm module. The first commit reflects the unaltered changes from rustfmt, and the commit message contains some notes on areas where I thought rustfmt had behaved strangely. The second commit attempts to fix the strange areas from the first commit.

Clarification edit: there are still some areas where I think rustfmt has made changes which may merit discussion (one is noted in the comments below). My second commit only undoes the changes that I figured would not warrant discussion (based on my opinion of the style, which is of course subjective).

r? @nrc

### CODE REVIEW

File: src/libterm/terminfo/parm.rs [hunk 1]
```diff
@@ -293,29 +333,34 @@ pub fn expand(cap: &[u8], params: &[Param], vars: &mut Variables)
...
             PushParam => {
                 // params are 1-indexed
                 stack.push(mparams[match cur.to_digit(10) {
                     Some(d) => d as usize - 1,
-                    None => return Err("bad param number".to_owned())
-                }].clone());
-            },
+                    None => return Err("bad param number".to_owned()),
+                }]
+                               .clone());
```
Comment (gkoz) [inline src/libterm/terminfo/parm.rs]: https://github.com/nrc/rustfmt/issues/426

**Comment by nrc** [comment]
r+, you'll need to add `#![feature(custom_attribute)]` and `#![allow(unused_attributes)]` to the crate root (lib.rs) to pass tests.

**Comment by SingingTree** [comment]
Cool bananas, will follow up on that later today so those tests can get passing.

File: src/libterm/terminfo/parm.rs [hunk 2]
```diff
@@ -144,122 +138,167 @@ pub fn expand(cap: &[u8], params: &[Param], vars: &mut Variables)
...
+                    },
                     '-' => if stack.len() > 1 {
                         match (stack.pop().unwrap(), stack.pop().unwrap()) {
                             (Number(y), Number(x)) => stack.push(Number(x - y)),
-                            _ => return Err("non-numbers on stack with -".to_owned())
+                            _ => return Err("non-numbers on stack with -".to_owned()),
                         }
-                    } else { return Err("stack is empty".to_owned()) },
+                    } else {
+                        return Err("stack is empty".to_owned())
```
Comment (marcusklaas) [inline src/libterm/terminfo/parm.rs]: rustfmt now appends these statements with semicolons; may be worthwhile to rerun this?

File: src/libterm/terminfo/parser/compiled.rs [hunk 3]
```diff
@@ -49,12 +52,14 @@ pub static numfnames: &'static[&'static str] = &[ "columns", "init_tabs", "lines
...
+#[rustfmt_skip]
 #[allow(missing_docs)]
 pub static numnames: &'static[&'static str] = &[ "cols", "it", "lines", "lm", "xmc", "pb",
     "vt", "wsl", "nlab", "lh", "lw", "ma", "wnum", "colors", "pairs", "ncv", "bufsz", "spinv",
     "spinh", "maddr", "mjump", "mcs", "mls", "npins", "orc", "orl", "orhi", "orvi", "cps", "widcs",
     "btns", "bitwin", "bitype", "UTug", "OTdC", "OTdN", "OTdB", "OTdT", "OTkn"];
 
+#[rustfmt_skip]
 #[allow(missing_docs)]
 pub static stringfnames: &'static[&'static str] = &[ "back_tab", "bell", "carriage_return",
```
Comment (marcusklaas) [inline src/libterm/terminfo/parser/compiled.rs]: Rustfmt should also be able to properly format these statics now

File: src/libterm/terminfo/parm.rs (hunk 1 above)
Comment (marcusklaas) [inline src/libterm/terminfo/parm.rs]: Fixed!

**Comment by SingingTree** [comment]
Cheers for the heads up. I will look at rerunning rustfmt and updating this PR.

**Comment by SingingTree** [comment]
I've merged master back into this branch and removed the rustfmt_skips and rerun rustfmt over the last commit in this branch. I don't see any particularly odd changes, and the statics are formatted without breaking the line limit. However it does result in some very long statics. I'd personally prefer that statics to remain truncated for the sake of avoiding hundreds of lines of them. Would be interested in other's thoughts on the matter.

**Comment by SingingTree** [comment]
It looks like this lastest batch of formatting causes a compilation failure in compiled.rs, I will investigate.

File: src/libterm/terminfo/parser/compiled.rs [hunk 4]
```diff
@@ -291,19 +735,19 @@ pub fn parse(file: &mut Read, longnames: bool)
...
-                },
+                                      string_table[offset as usize..(offset as usize + len)]
+                                          .to_vec())
+                }
                 None => {
-                    return Err("invalid file: missing NUL in \
-                                string_table".to_owned());
+                    return Err("invalid file: missing NUL in string_table".to_owned());
                 }
-            };
```
Comment (SingingTree) [inline src/libterm/terminfo/parser/compiled.rs]: It looks like removal of the semi-colon here by rustfmt causes breaking changes

**Comment by SingingTree** [comment]
Looks like I've bungled this a bit by attempting to merge master with the already made formatting. I think that's resulted in some issues. So I'm just going to reset this branch to master and repeat the formatting.

File: src/libterm/terminfo/parser/compiled.rs [hunk 5]
```diff
@@ -291,19 +735,19 @@ pub fn parse(file: &mut Read, longnames: bool)
...
+            }
```
Comment (SingingTree) [inline src/libterm/terminfo/parser/compiled.rs]: I think the removal of the semi-colon here causes issues.

**Comment by SingingTree** [comment]
Updated to work with upstream changes. Changes made reflect rustfmt's changes, aside from the skips in compiled.rs to avoid splitting the statics over many more lines.

**Pull request merged**